templates/windows/main.cpp

```diff
@@ -25,34 +25,76 @@ you.
 ****************************************************************************/
 
 #include <iostream>
+#include <windows.h>
 
 #include "platform/BasePlatform.h"
 
 #if defined(CC_SERVER_MODE)
-#include <windows.h>
-int WINAPI WinMain(HINSTANCE hInst, HINSTANCE hPrev, LPSTR szCmdLine, int sw) {
-  if (!AllocConsole()) {
-    // GetLastError() to get more info about the error.
-    MessageBox(NULL, L"The console window was not created", NULL,
-               MB_ICONEXCLAMATION);
-    return 0;
-  }
-
-  FILE *fConsole;
-  freopen_s(&fConsole, "CONOUT$", "w", stdout);
-  freopen_s(&fConsole, "CONOUT$", "w", stderr);
-  freopen_s(&fConsole, "CONIN$", "r", stdin);
-
-  START_PLATFORM(0, nullptr);
-
-  fclose(fConsole);
-  if (!FreeConsole())
-    MessageBox(NULL, L"Failed to free the console!", NULL, MB_ICONEXCLAMATION);
+
+static bool UTF16ToUTF8(LPCWSTR wideStr, const char** outUtf8) {
+    if (outUtf8 == nullptr) {
+        return false;
+    }
+    int utf8BufferBytes = ::WideCharToMultiByte(CP_UTF8, 0, wideStr, -1, NULL, 0, NULL, NULL);
+    if (utf8BufferBytes < 0) {
+        outUtf8 = nullptr;
```

Comment:
*outUtf8 = nullptr?

Reply:
Yes, the output parameter is a secondary pointer `const char**`.
If there is an error, reset the `*outUtf8` (const char*) to nullptr.

Reply:
I see, so the code is error here, it set `outUtf8` instead of `*outUtf8`.

Reply:
Yep, I made the mistake. I think I needs to add one more `const` for that.

Reply:
fixed and set the type of outUtf8 parameter to `const char** const outUtf8`

Reply:
fixed and set the type of outUtf8 parameter to `const char** const outUtf8`